tests/testthat/test-subscript-loc.R

```diff
@@ -237,11 +237,18 @@ test_that("character subscripts require named vectors", {
 
 test_that("can optionally extend beyond the end", {
   expect_error(num_as_location(1:5, 3), class = "vctrs_error_subscript_oob")
+
   expect_identical(num_as_location(1:5, 3, oob = "extend"), 1:5)
+  expect_identical(num_as_location(4:5, 3, oob = "extend"), 4:5)
+  expect_identical(num_as_location(6:4, 3, oob = "extend"), 6:4)
   expect_identical(num_as_location(c(1:5, 7, 6), 3, oob = "extend"), c(1:5, 7L, 6L))
-  expect_identical(num_as_location(c(1, NA, 3), 2, oob = "extend"), c(1L, NA, 3L))
+  expect_identical(num_as_location(c(1, NA, 4, 3), 2, oob = "extend"), c(1L, NA, 4L, 3L))
```

Comment:
Can you move this to a separate `"can extend beyond the end consecutively but non-monotonically"` test please?